server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistTestsStepTest.java

```diff
@@ -0,0 +1,344 @@
+    BatchReportWriter writer = new BatchReportWriter(reportDir);
+    List<BatchReport.Test> batchTests = Arrays.asList(newTest(1), newTest(2));
+    writer.writeTests(TEST_FILE_REF_1, batchTests);
+
+    sut.execute(new ComputationContext(new BatchReportReader(reportDir), newProjectDto(PROJECT_UUID)));
+
+    FileSourceDto dto = dbClient.fileSourceDao().selectTest(TEST_FILE_UUID_1);
+    assertThat(dto.getFileUuid()).isEqualTo(TEST_FILE_UUID_1);
+    List<FileSourceDb.Test> tests = dto.getTestData();
+    assertThat(tests).hasSize(2);
+    assertThat(tests.get(0).getCoveredFileList()).isEmpty();
+  }
+
+  @Test
+  public void insert_coverage_details_not_taken_into_account() throws Exception {
+    BatchReportWriter writer = new BatchReportWriter(reportDir);
+    List<BatchReport.Test> batchTests = Arrays.asList(newTest(1));
+    writer.writeTests(TEST_FILE_REF_1, batchTests);
+    List<CoverageDetail> coverageDetails = Arrays.asList(newCoverageDetail(1, MAIN_FILE_REF_1), newCoverageDetail(2, MAIN_FILE_REF_2));
+    writer.writeCoverageDetails(TEST_FILE_REF_1, coverageDetails);
+    writer.writeCoverageDetails(TEST_FILE_REF_2, coverageDetails);
+
+    sut.execute(new ComputationContext(new BatchReportReader(reportDir), newProjectDto(PROJECT_UUID)));
+
+    assertThat(log.logs(LoggerLevel.WARN)).hasSize(1);
+    assertThat(log.logs(LoggerLevel.WARN).get(0)).isEqualTo("Some coverage tests are not taken into account during analysis of project 'KEY_PROJECT'");
+    assertThat(log.logs(LoggerLevel.TRACE)).hasSize(2);
+    assertThat(log.logs(LoggerLevel.TRACE).get(0)).isEqualTo("The following test coverages for file 'TEST-PATH-1' have not been taken into account: name#2");
+    assertThat(log.logs(LoggerLevel.TRACE).get(1)).startsWith("The following test coverages for file 'TEST-PATH-2' have not been taken into account: ");
+    assertThat(log.logs(LoggerLevel.TRACE).get(1)).contains("name#1", "name#2");
+  }
+
+  @Test
+  public void aggregate_coverage_details() throws Exception {
+    BatchReportWriter writer = new BatchReportWriter(reportDir);
+    writer.writeTests(TEST_FILE_REF_1, Arrays.asList(newTest(1)));
+    writer.writeCoverageDetails(TEST_FILE_REF_1, Arrays.asList(
+      newCoverageDetailWithLines(1, MAIN_FILE_REF_1, 1, 3),
+      newCoverageDetailWithLines(1, MAIN_FILE_REF_1, 2, 4)));
+
+    sut.execute(new ComputationContext(new BatchReportReader(reportDir), newProjectDto(PROJECT_UUID)));
+
+    FileSourceDto dto = dbClient.fileSourceDao().selectTest(TEST_FILE_UUID_1);
+    List<Integer> coveredLines = dto.getTestData().get(0).getCoveredFile(0).getCoveredLineList();
+    assertThat(coveredLines).containsOnly(1, 2, 3, 4);
+  }
+
+  @Test
+  public void update_one_test() throws Exception {
+    // ARRANGE
+    dbClient.fileSourceDao().insert(session, new FileSourceDto()
+      .setProjectUuid(PROJECT_UUID)
+      .setFileUuid(TEST_FILE_UUID_1)
+      .setTestData(Arrays.asList(newDbTest(1)))
```

Comment:
It's not clear which fields has been updated -> instead of using newDbTest(1), could you create the dto in the test -> FileSourceDb.Test.newBuilder()
      .setName("name#" + id)...